templates/template/scripts/pull-scripts

```diff
@@ -5,19 +5,37 @@ cd $(dirname $0)
 
 source ./version
 
+if ls ../bin/charts-build-scripts 1>/dev/null 2>/dev/null; then
+    CURRENT_SCRIPT_VERSION=$(../bin/charts-build-scripts --version | cut -d' ' -f3)
+    if [[ "${CURRENT_SCRIPT_VERSION}" == "${CHARTS_BUILD_SCRIPT_VERSION}" ]]; then
+        exit 0
+    fi
+fi
+
 echo "Pulling in charts-build-scripts version ${CHARTS_BUILD_SCRIPTS_REPO}@${CHARTS_BUILD_SCRIPT_VERSION}"
 
 rm -rf ../bin
 cd ..
 
-rm -rf charts-build-scripts
-git clone --depth 1 --branch $CHARTS_BUILD_SCRIPT_VERSION $CHARTS_BUILD_SCRIPTS_REPO 2>/dev/null
+mkdir -p bin
+ARCH=$(go version | cut -d' ' -f4 | cut -d'/' -f1)
+curl -s -L ${CHARTS_BUILD_SCRIPTS_REPO%.git}/releases/download/${CHARTS_BUILD_SCRIPT_VERSION}/charts-build-scripts-${ARCH} --output bin/charts-build-scripts
+if [[ $(cat bin/charts-build-scripts) == "Not Found" ]]; then 
```

Comment:
I see the following message when charts-build-scripts does not exist on Mac:

```
> cat bin/charts-build-scripts
cat: bin/charts-build-scripts: No such file or directory
```

Reply:
I'm a little confused about what you mean. The previous step curls the binary into `bin/charts-build-scripts`.

Maybe the fix for this is:

```suggestion
if ! [[ -f bin/charts-build-scripts ]] || [[ $(cat bin/charts-build-scripts) == "Not Found" ]]; then 
```

Reply:
Are we comparing the output of the bash command  `cat bin/charts-build-scripts` with the string `Not Found` here? 

In my test, when `bin/charts-build-scripts` does not exist, the output of  `cat bin/charts-build-scripts` command is 
```
cat: bin/charts-build-scripts: No such file or directory
```
not `Not Found`

Reply:
Yes, but by definition `bin/charts-build-scripts` should exist if the preceding curl command was successful since we specify `--output bin/charts-build-scripts`.

Reply:
I committed the fix suggested in https://github.com/rancher/charts-build-scripts/pull/42#discussion_r641689874, although I didn't seem to need it in local setup; could you try again?

Reply:
Oh, now I understand what it does.  `Not Found` is the output of the curl command and is save as charts-build-scripts 😆